app/src/main/java/io/appium/settings/RecorderThread.java

```diff
@@ -0,0 +1,419 @@
+                    return;
+                }
+                try {
+                    while (!audioStopped) {
+                        int index = audioEncoder.dequeueInputBuffer(
+                                RecorderConstant.MEDIA_QUEUE_BUFFERING_DEFAULT_TIMEOUT);
+                        if (index < 0) {
+                            continue;
+                        }
+                        ByteBuffer inputBuffer = audioEncoder.getInputBuffer(index);
+                        if (inputBuffer == null) {
+                            if (!stopped) {
+                                asyncError = true;
+                            }
+                            return;
+                        }
+                        inputBuffer.clear();
+                        int read = audioRecord.read(inputBuffer, inputBuffer.capacity());
+                        if (read < 0) {
+                            if (!stopped) {
+                                asyncError = true;
+                            }
+                            break;
+                        }
+                        audioEncoder.queueInputBuffer(index, 0, read,
+                                getPresentationTimeUs(), 0);
+                    }
+                } catch (Exception e) {
+                    if (!stopped) {
+                        Log.e(TAG, "Audio error", e);
+                        asyncError = true;
+                        e.printStackTrace();
+                    }
+                } finally {
+                    audioRecord.stop();
+                    audioRecord.release();
+                    audioRecord = null;
+                }
+            }
+        });
+        audioRecordThread.start();
+    }
+
+    private long getPresentationTimeUs() {
+        if (!startTimestampInitialized) {
+            startTimestampUs =
+                    System.nanoTime() / RecorderConstant.ENCODER_PRESENTATION_TIME_DEFAULT_DIVIDER;
+            startTimestampInitialized = true;
+        }
+        return (System.nanoTime() / RecorderConstant.ENCODER_PRESENTATION_TIME_DEFAULT_DIVIDER
+                        - startTimestampUs);
+    }
+
+    private int calcBitRate(int width, int height) {
+        final int bitrate = (int) (RecorderConstant.BITRATE_MULTIPLIER *
+                RecorderConstant.VIDEO_CODEC_FRAME_RATE * width * height);
+        Log.i(TAG, String.format("bitrate=%5.2f[Mbps]", bitrate / 1024f / 1024f));
+        return bitrate;
+    }
+
+    private void startMuxerIfSetUp() {
+        if (audioTrackIndex >= 0 && videoTrackIndex >= 0) {
+            muxer.start();
+            muxerStarted = true;
+        }
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    private boolean writeAudioBufferToFile() {
+        int encoderStatus;
+
+        encoderStatus = audioEncoder.dequeueOutputBuffer(this.bufferInfo, 0);
+        if (encoderStatus == MediaCodec.INFO_OUTPUT_FORMAT_CHANGED) {
+            if (audioTrackIndex > 0) {
```

Comment:
this comparison does not match to the log message below

Reply:
done